apollo-router/src/plugins/fleet_detector.rs

```diff
@@ -473,3 +469,270 @@ fn get_deployment_type() -> &'static str {
+                MetricType::Counter,
+                &[KeyValue::new("subgraph.service.name", "subgraph"),],
+            ));
+            assert!(!collect_metrics().metric_exists::<u64>(
+                "apollo.router.operations.fetch.response_size",
+                MetricType::Counter,
+                &[KeyValue::new("subgraph.service.name", "subgraph"),],
+            ));
+        }
+        .with_metrics()
+        .await;
+    }
+
+    #[tokio::test]
+    async fn test_enabled_http_client_service() {
+        async {
+            // WHEN the plugin is enabled
+            let plugin = FleetDetector {
+                enabled: true,
+                ..Default::default()
+            };
+
+            // GIVEN an http client service request
+            let mut mock_bad_request_service = MockHttpClientService::new();
+            mock_bad_request_service.expect_call().times(1).returning(
+                |req: http::Request<Body>| {
+                    Box::pin(async {
+                        let data = hyper::body::to_bytes(req.into_body()).await?;
+                        Ok(http::Response::builder()
+                            .status(StatusCode::BAD_REQUEST)
+                            .header("content-type", "application/json")
+                            // making sure the request body is consumed
+                            .body(Body::from(data))
+                            .unwrap())
+                    })
+                },
+            );
+            let mut bad_request_http_client_service = plugin.http_client_service(
+                "subgraph",
+                mock_bad_request_service
+                    .map_request(|req: HttpRequest| req.http_request.map(|body| body.into_inner()))
+                    .map_response(|res: http::Response<Body>| HttpResponse {
+                        http_response: res.map(RouterBody::from),
+                        context: Default::default(),
+                    })
+                    .boxed(),
+            );
+            let http_client_req = HttpRequest {
+                http_request: http::Request::builder()
+                    .body(RouterBody::from("request"))
+                    .unwrap(),
+                context: Default::default(),
+            };
+            let http_client_response = bad_request_http_client_service
+                .ready()
+                .await
+                .unwrap()
+                .call(http_client_req)
+                .await
+                .unwrap();
+
+            // making sure the response body is consumed
+            let _data = hyper::body::to_bytes(http_client_response.http_response.into_body())
+                .await
+                .unwrap();
+
+            // THEN fetch metrics should exist
+            assert_counter!(
+                "apollo.router.operations.fetch",
+                1,
+                &[
+                    KeyValue::new("subgraph.service.name", "subgraph"),
```

Comment:
Nit: can we make sure to use `subgraph.name` instead to be consistent with what we have in the [official subgraph attributes](https://www.apollographql.com/docs/graphos/reference/router/telemetry/instrumentation/standard-attributes#subgraph)